Are the shortlogs useful - yes .. they catch what appear to be mistakes

Specifically: What happened to the aacraid ioctl security fix ? Did someone decide it
wasn't needed or did it get lost somewhere on the way ?

While this looks scary the only obvious exploit cases are where the user can
open a device level file on an AACraid. Very few people put scanners or CD
devices on one so the actual impact is probably minimal.

Alan

--

Signed-off-by: Alan Cox <alan@redhat.com>

--- a/drivers/scsi/aacraid/linit.c	2007-07-09 10:51:55.653223304 +0100
+++ b/drivers/scsi/aacraid/linit.c	2007-07-09 10:51:55.653223304 +0100
@@ -453,6 +453,8 @@
 static int aac_ioctl(struct scsi_device *sdev, int cmd, void __user * arg)
 {
 	struct aac_dev *dev = (struct aac_dev *)sdev->host->hostdata;
+	if (!capable(CAP_SYS_RAWIO))
+		return -EPERM;
 	return aac_do_ioctl(dev, cmd, arg);
 }
 
@@ -645,6 +647,8 @@
 static int aac_compat_ioctl(struct scsi_device *sdev, int cmd, void __user *arg)
 {
 	struct aac_dev *dev = (struct aac_dev *)sdev->host->hostdata;
+	if (!capable(CAP_SYS_RAWIO))
+		return -EPERM;
 	return aac_compat_do_ioctl(dev, cmd, (unsigned long)arg);
 }
 
